### DISCUSSION TITLE
Confirmation email not received

### DISCUSSION THREAD

**Discussion by joossii** [padloc/padloc#439] [Q&A] [answered]
Ive installed padloc with the instructions on padloc website but i cannot get the mail to work. So i cannot register a new account because the email with the code is never received. 

I have tried everything, google email, sendgrid, own domain email. No email is being sent. When i test the smtp it works, but when i put in the details in the .env file .... nothing. It says it send it, but ... nothing received. I can't check the log because once installed like said on padloc website selfhosting there is no log directory made ....

**Comment by MaKleSoft** [comment]
Hi @joossii, sorry for the late response! Can you please post the output of the server process (stdout)?

**Reply by Mohit-3430** [reply]
Even I am facing the same issue:

Following file : [sample.env](https://github.com/padloc/padloc/blob/v4/docs/examples/config/example.env)

Here is the stdout : 
```
@padloc/server: Err [Error]: Failed to send email to ************@gmail.com
@padloc/server:     at EmailAuthServer.initAuthRequest (E:\Projects\padloc\packages\core\src\auth\email.ts:58:19)    
@padloc/server:     at processTicksAndRejections (node:internal/process/task_queues:96:5)
@padloc/server:     at C.startAuthRequest (E:\Projects\padloc\packages\core\src\server.ts:373:29)
@padloc/server:     at C.process (E:\Projects\padloc\packages\core\src\server.ts:236:24)
@padloc/server:     at Server.handle (E:\Projects\padloc\packages\core\src\server.ts:1917:31)
@padloc/server:     at Server.<anonymous> (E:\Projects\padloc\packages\server\src\transport\http.ts:77:37) {
@padloc/server:   time: 2022-05-08T10:01:11.187Z,
@padloc/server:   code: 'email_verification_failed',
@padloc/server:   report: false,
@padloc/server:   display: false,
@padloc/server:   originalError: undefined
@padloc/server: }
```

**Reply by jpapach** [reply]
Same. Checked SPF, DKIM, and DMARC. Disabled all temporarily and tried from multiple mail servers and service providers.

**Reply by MaKleSoft** [reply]
@Mohit-3430 Based on the stdout you posted, you're working with the v4 branch, correct? @joossii @jpapach Which version are you working with?

Instructions for v4 are slightly different from v3 and can be found here: https://github.com/padloc/padloc/tree/v4/docs/examples/hosting/docker

For reference, here is the v3 guide: https://padloc.app/help/guides/self-host/

Sorry for the patchy documentation! We'll consolidate and expand the docs once we've merged v4.

**Reply by Mohit-3430** [reply]
Yeah @MaKleSoft I am on v4, no problem as long as u are with us we will not have issues unsolved, yeah please improve documentation, thanks .

**Reply by Mohit-3430** [reply]
But I got the same stdout again after following the instructions.

**Reply by jpapach** [reply]
V4 here as well. Will wait for documentation updates.

**Reply by BrunoBernardino** [reply]
@jpapach are you also on Windows? (the instructions at https://github.com/padloc/padloc/tree/v4/docs/examples/hosting/docker should work and we're not seeing issues on Linux nor macOS)

**Reply by jpapach** [reply]
@BrunoBernardino No sir. Installed on Ubuntu 20.04.4 LTS.

**Reply by BrunoBernardino** [reply]
@jpach Thanks! Can you add a `console.log(e)` at https://github.com/padloc/padloc/blob/v4/packages/core/src/auth/email.ts#L57-L59 and see if there's any more useful information?

@MaKleSoft is there a reason not to log errors like that with `console.log` or `log` from `logging.ts` (since those can be very helpful for people self-hosting)?

**Reply by jpapach** [reply]
@BrunoBernardino You'll have to bear with me as I am not overly familiar with Linux commands, other than following instructions as given. If you could provide the steps needed to add console.log(e) I will do so. If however, you think that I should know how to do this already in order to request help with this issue, I understand and will wait to see if a solution is presented to someone with more knowledge than myself.

**Reply by BrunoBernardino** [reply]
No worries @jpapach ! In that file in your local system (`packages/core/src/auth/email.ts`), where you see those lines (57 to 59), add `console.log(e);` above the `throw ...` one, and restart your server and try requesting the confirmation email. The error logs should now show some more information.

**Reply by jpapach** [reply]
@BrunoBernardino That may be the issue. When I run (find . -type f -name "email.ts") I get no results. I found the path buried in docker where I think it should be, but the file isn't there.

**Reply by BrunoBernardino** [reply]
Sorry @jpapach , I didn't realize you were using the docker image. You won't have that file in your local filesystem, it'll be inside the docker container. If you're feeling brave, you can copy the file to your local filesystem, edit it, and copy it back, before restarting the container:

- `docker cp <container-name>:/padloc/packages/core/src/auth/email.ts .` to copy the file to your local directory/filesystem
- edit the local `email.ts` file
- `docker cp email.ts <container-name>:/padloc/packages/core/src/auth/` to copy it back into the container

If that's too much, it's fine. I can instead prepare a separate `docker-compose.yml` file for you with a specific branch with the changes I'm talking about, to figure out what might be happening to you.

**Reply by jpapach** [reply]
@BrunoBernardino Might need that .yml. Worked half the night trying to get that file off of the container with no luck. Thank you for your assistance.

**Reply by BrunoBernardino** [reply]
Ok, no worries. I'll try to provide that later today.

**Reply by BrunoBernardino** [reply]
@jpapach since you probably have customizations in your own `docker-compose.yml`, just look for the lines that have `context: github.com/padloc/padloc.git#v4` (should be two different lines) and replace that with `context: github.com/padloc/padloc.git#debug/email-logging`. Let me know if that works for you and if there is more information in the error now.

**Reply by Mohit-3430** [reply]
> No worries @jpapach ! In that file in your local system (`packages/core/src/auth/email.ts`), where you see those lines (57 to 59), add `console.log(e);` above the `throw ...` one, and restart your server and try requesting the confirmation email. The error logs should now show some more information.

In My case this is the output of console.log();

```
@padloc/server: [Error: 140018586560448:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:
@padloc/server: ] {
@padloc/server:   library: 'SSL routines',
@padloc/server:   function: 'ssl3_get_record',
@padloc/server:   reason: 'wrong version number',
@padloc/server:   code: 'ESOCKET',
@padloc/server:   command: 'CONN'
@padloc/server: }
```

**Reply by BrunoBernardino** [reply]
Now we're getting somewhere! So it's an SSL problem. @MaKleSoft any insights on what to look at here? Could we be missing something in the documentation about the SSL setup?

**Reply by MaKleSoft** [reply]
@Mohit-3430 @jpapach Which port are trying to connect on? If you're connecting on `587` or `465`, you may need to set `PL_EMAIL_SMTP_SECURE=true`. If that doesn't work, try `PL_EMAIL_SMTP_IGNORE_TLS=true`.

**Reply by jpapach** [reply]
@MaKleSoft I've tried both, with the secure flag set. Next, I'll try with the ignore function next, and then a simple port 25 with the secure flag set to false to see what happens.

**Reply by Mohit-3430** [reply]
> @Mohit-3430 @jpapach Which port are trying to connect on? If you're connecting on `587` or `465`, you may need to set `PL_EMAIL_SMTP_SECURE=true`. If that doesn't work, try `PL_EMAIL_SMTP_IGNORE_TLS=true`.

I'm using 587 port and both flags set true.

**Reply by jpapach** [reply]
@MaKleSoft I've tried port 25 with no secure flag, 465 with secure flag set, 465 with ignore tls, 587 with secure flag, and 587 with ignore tls. All of which result in no emails being sent when signing up for an account. I have checked the mail server logs for incoming connections, and nothing is making it to the server. Not even failed login attempts from the account. I have verified that the padloc server can ping the mail server IP and can resolve the DNS name corectly. I hope @Mohit-3430  has better luck. Thank you for all of your help.

**Comment by Mohit-3430** [comment, marked as answer]
So yeah finally worked!! I changed Nothing in the config or .env files, only change in my case is :

I am using outlook email so added the following line in `smtp.ts`
`service: "Outlook365",` (we can add accordingly based on our smtp provider)
we can add that anywhere in nodemailer's creatingTransport() method

Got helped with the message from logger (console) then found this [link](https://stackoverflow.com/questions/66317125/node-js-nodemailer-error-wrong-version-number-invalid-greeting) useful. Thanks for being there. 👍

**Reply by jpapach** [reply]
@Mohit-3430 Thanks for that! In my case, the server is Kerio Connect. service: "Kerio Connect", made it work. Still don't know why the Gmail wasn't working. In any event, the issue is resolved!

**Reply by BrunoBernardino** [reply]
Oh, that's so great to know, thanks for sharing the solution!

**Reply by joossii** [reply]
And how does it work for normal smtp then? I dont have connect nor 365 outlook

**Reply by BrunoBernardino** [reply]
@joossii do you receive the same `routines:ssl3_get_record:wrong version number` error?
